src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -413,6 +414,38 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
 
   import syntaxAnalyzer.{ UnitScanner, UnitParser, JavaUnitParser }
 
+  private def resolveSettingImports(settingValue: String, settingName: String): List[Import] = {
```

Comment:
Got it. I have to use a type projection `ScalaSettings#StringSetting` instead of `settings.StringSetting`, otherwise scalac laments about it not being a stable identifier.